Terminal.Gui/Views/CollectionNavigation/DefaultCollectionNavigatorMatcher.cs

```diff
@@ -0,0 +1,30 @@
+﻿#nullable enable
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Default implementation of <see cref="ICollectionNavigatorMatcher"/>, performs
+///     case-insensitive (see <see cref="Comparer"/>) matching of items based on
+///     <see cref="object.ToString()"/>.
+/// </summary>
+public class DefaultCollectionNavigatorMatcher : ICollectionNavigatorMatcher
```

Comment:
It is interesting that this entire implementation could actually be done as default on the interface. Then the interface could declare the comparer as only getter default property and the class could have an init only member for the comparer. If my mental compilation is correct this would work